pkg/envoy/registry/registry.go

```diff
@@ -19,27 +17,16 @@ func NewProxyRegistry(mapper ProxyServiceMapper, msgBroker *messaging.Broker) *P
 
 // RegisterProxy implements MeshCatalog and registers a newly connected proxy.
 func (pr *ProxyRegistry) RegisterProxy(proxy *envoy.Proxy) {
-	pr.connectedProxies.Store(proxy.GetCertificateCommonName(), connectedProxy{
+	pr.connectedProxies.Store(proxy.UUID.String(), connectedProxy{
 		proxy:       proxy,
 		connectedAt: time.Now(),
 	})
-
-	// If this proxy object is on a Kubernetes Pod - it will have an UID
-	if proxy.HasPodMetadata() {
-		podUID := types.UID(proxy.PodMetadata.UID)
-
-		// Create a PodUID to Certificate CN map so we can easily determine the CN from the PodUID
-		pr.podUIDToCN.Store(podUID, proxy.GetCertificateCommonName())
-
-		// Create a PodUID to Cert Serial Number so we can easily look-up the SerialNumber of the cert issued to a proxy for a given Pod.
-		pr.podUIDToCertificateSerialNumber.Store(podUID, proxy.GetCertificateSerialNumber())
-	}
```

Comment:
I'd also note in the commit description that the change removes dead code arising from the refactor.